lib/client-assets.php

```diff
@@ -1147,19 +1147,28 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 	// Initialize the editor.
 	$gutenberg_theme_support = get_theme_support( 'gutenberg' );
 	$align_wide              = get_theme_support( 'align-wide' );
-	$color_palette           = get_theme_support( 'editor-color-palette' );
+	$color_palette           = current( (array) get_theme_support( 'editor-color-palette' ) );
+
+	// Backcompat for Color Palette set as multiple parameters.
+	if ( is_string( $color_palette ) || isset( $color_palette['color'] ) ) {
+		$color_palette = get_theme_support( 'editor-color-palette' );
```

Comment:
This is a solid implementation of BC, just a minor detail I think we should improve: I don't like calling `get_theme_support()` twice. I think we should make the original call to set `$color_palette` be only the `(array) get_theme_support( 'editor-color-palette' )` part.

Then the if clause could check `! empty( $color_palette ) && isset( $color_palette[0]['color'] )`, and if so, trigger the notice. Then an else clause could get the first array element to account for the new way.

One thing I'm not sure about, why is there an `is_string()` check. I might miss something, was it possible to provide strings only at some point?

Reply:
@felixarntz 
From my tests, we can not use `(array) get_theme_support( 'editor-color-palette' )` width `! empty( $color_palette )` check as if the theme does not claim the support, this would be returned (which is not en empty array):

```php
array(1) {
	[0] => bool(false)
}
```

The `is_string()` check is there to accommodate the old way of setting colors as `add_theme_support( 'editor-color-palette', '#fff', '#000', ... );`. Then this was updated to setting colors with multiple array arguments.

This code should do the trick though:
```php
	$color_palette           = (array) get_theme_support( 'editor-color-palette' );

	// Backcompat for Color Palette set as multiple parameters.
	if ( is_string( $color_palette[0] ) || isset( $color_palette[0]['color'] ) ) {
		_doing_it_wrong(
			'add_theme_support()',
			__( 'Setting colors using multiple parameters is deprecated. Please pass a single parameter with an array of colors. See https://wordpress.org/gutenberg/handbook/extensibility/theme-support/ for details.', 'gutenberg' ),
			'3.4.0'
		);
	} else {
		$color_palette = current( $color_palette );
	}
```

Reply:
@felixarntz I've went ahead and created #7695 addressing your concerns. Thanks for pointing this out!